### PR TITLE
[Auto Parallel] Merge the python and c++ impls of ProcessMesh

### PR DESCRIPTION
**Pull request by aoyulong**
### PR types

Others

### PR changes

Others

### Describe

- Merge the python and c++ impls
    - The python ProcessMesh is a wrapper of the c++ ProcessMesh (by inheritance).
- Rename the two methods of `ProcessMesh`:
    - `topology` -> `shape`
    - `processes` -> `process_ids`

### CODE REVIEW

File: paddle/fluid/pybind/auto_parallel_py.cc [hunk 1]
```diff
@@ -131,7 +137,7 @@ void BindAutoParallel(py::module *m) {
                     const std::vector<std::string> &>(),
            py::arg("name"),
            py::arg("shape"),
-           py::arg("process_ids"),
+           py::arg("device_ids"),
```
Comment (XiaoguangHu01) [inline paddle/fluid/pybind/auto_parallel_py.cc:140]: 这些接口对外开放了吗？这处修改是否有兼容性风险？

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: paddle/fluid/pybind/auto_parallel_py.cc (hunk 1 above)
Comment (aoyulong) [reply paddle/fluid/pybind/auto_parallel_py.cc:140]: 暂时不对外开放。

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**